### PR TITLE
Added IO_push

### PR DESCRIPTION
**Pull request by jasonlshelton**
added function to push a data point to an adafruitio.com feed. Creates feed if it does not exist

### CODE REVIEW

**Comment by siddacious** [comment]
Thanks @jasonlshelton! It looks like travis is failing due to some PyLint errors. Please take a look and let us know if you need help resolving them.

**Comment by jasonlshelton** [comment]
I could use some guidance in resolving the current issue that is failing the check.

File: adafruit_pyportal.py [hunk 1]
```diff
@@ -649,6 +654,47 @@ def image_converter_url(image_url, width, height, color_depth=16):
                                           width, height,
                                           color_depth, image_url)
 
+    def io_push(self, feed, data):
```
Comment (brentru) [inline adafruit_pyportal.py]: Could you change `feed` to `feed_key`?

File: adafruit_pyportal.py [hunk 2]
```diff
@@ -649,6 +654,47 @@ def image_converter_url(image_url, width, height, color_depth=16):
...
+        # pylint: disable=line-too-long
+        """Push data to an adafruit.io feed
+
+        :param str feed: Name of feed to push data to.
```
Comment (brentru) [inline adafruit_pyportal.py]: Feed key to send data to.

File: adafruit_pyportal.py [hunk 3]
```diff
@@ -649,6 +654,47 @@ def image_converter_url(image_url, width, height, color_depth=16):
...
+        :param data: data to send to feed
+
+        """
+        # pylint: enable=line-too-long
+
+        try:
+            aio_username = secrets['aio_username']
+            aio_key = secrets['aio_key']
+        except KeyError:
+            raise KeyError("\n\n")
```
Comment (brentru) [inline adafruit_pyportal.py]: Add a more verbose with the `KeyError`, something like "Adafruit IO secrets are kept in secrets.py, please add them there!" works

File: adafruit_pyportal.py [hunk 4]
```diff
@@ -649,6 +654,47 @@ def image_converter_url(image_url, width, height, color_depth=16):
...
+
+        wifi = adafruit_esp32spi_wifimanager.ESPSPI_WiFiManager(self._esp, secrets, None)
+        io_connect = RESTClient(aio_username, aio_key, wifi)
```
Comment (brentru) [inline adafruit_pyportal.py]: Change `io_connect` to `io_client`

File: adafruit_pyportal.py [hunk 5]
```diff
@@ -649,6 +654,47 @@ def image_converter_url(image_url, width, height, color_depth=16):
...
+
+        while True:
+            try:
+                feed_id = io_connect.get_feed(feed)
+            except AdafruitIO_RequestError:
+                # If no feed exists, create one
+                feed_id = io_connect.create_new_feed(feed)
+            except RuntimeError as exception:
```
Comment (brentru) [inline adafruit_pyportal.py]: Only catching `AdafruitIO_RequestError` is required.

**Comment by jasonlshelton** [comment]
I've made the style changes you requested. How does it look?

**Pull request closed without merging**